src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.directory.server;
+
+import org.gridsuite.directory.server.dto.AccessRightsAttributes;
+import org.gridsuite.directory.server.dto.DirectoryAttributes;
+import org.gridsuite.directory.server.dto.ElementAttributes;
+import org.gridsuite.directory.server.repository.DirectoryElementEntity;
+import org.gridsuite.directory.server.repository.DirectoryElementRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * @author Nicolas Noir <nicolas.noir at rte-france.com>
+ */
+@ComponentScan(basePackageClasses = DirectoryElementRepository.class)
+@Service
+class DirectoryService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DirectoryService.class);
+
+    private final DirectoryElementRepository directoryElementRepository;
+
+    DirectoryService(DirectoryElementRepository directoryElementRepository) {
+        this.directoryElementRepository = directoryElementRepository;
+    }
+
+    private static ElementAttributes fromEntity(DirectoryElementEntity entity) {
+        return new ElementAttributes(entity.getId(), entity.getName(), ElementType.valueOf(entity.getType()), new AccessRightsAttributes(entity.isPrivate()), entity.getOwner());
+    }
+
+    public Mono<DirectoryElementEntity> createDirectory(DirectoryAttributes directoryAttributes) {
+        Mono<DirectoryElementEntity> createdDirectory = directoryElementRepository.save(new DirectoryElementEntity(null, directoryAttributes.getParentId(),
```

Comment:
you can immediately return the expression

Reply:
Corrected